sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -440,6 +440,20 @@ def _fit(self, X, y, incremental=False):
             self._fit_lbfgs(
                 X, y, activations, deltas, coef_grads, intercept_grads, layer_units
             )
+
+        # validate parameter weights
+        check_weights = (
+            lambda coefs, intercept: np.isfinite(coefs).all()
+            and np.isfinite(intercept).all()
+        )
+        if not np.all(
+            [
+                check_weights(coef, inter)
+                for coef, inter in zip(self.coefs_, self.intercepts_)
+            ]
+        ):
+            raise ValueError("Solver produced non-finite parameter weights.")
```

Comment:
Similar comment from https://github.com/scikit-learn/scikit-learn/pull/22149/files#r780722680 We can adjust the message so users know what they can do about it.

```suggestion
            raise ValueError("coefs or intercepts have non-finite parameter weights. "
                             "The input data may contain large values and need to be "
                             "preprocessed.")
```

Reply:
May you include the second part of the message?

Reply:
May you include the second part of the error message?